### PR TITLE
feat(tile-inspector): show original texture

### PR DESCRIPTION
**Pull request by mspivak-actionengine**

### CODE REVIEW

File: src/components/tile-details-panel/texture-section.tsx [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+  COMPRESSED_RED_GREEN_RGTC2_EXT,
+  COMPRESSED_SIGNED_RED_GREEN_RGTC2_EXT,
+  COMPRESSED_SRGB_S3TC_DXT1_EXT,
+  COMPRESSED_SRGB_ALPHA_S3TC_DXT1_EXT,
+  COMPRESSED_SRGB_ALPHA_S3TC_DXT3_EXT,
+  COMPRESSED_SRGB_ALPHA_S3TC_DXT5_EXT,
+} = GL_EXTENSIONS_CONSTANTS;
+
+// eslint-disable-next-line complexity
+const isFormatSupported = (gl, format) => {
```
Comment (belom88) [inline src/components/tile-details-panel/texture-section.tsx:73]: Move functions to utils. I think `src\utils\debug\texture-render-utils.ts` is a right place

File: src/components/tile-details-panel/texture-section.tsx [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+  width: ${({ width }) => `${width}px`};
+  position: relative;
+  margin: 0;
+  border: 0;
+  background-image: ${({ image }) => `${image}`};
+  background-repeat: no-repeat;
+  cursor: inherit;
+`;
+
+const createCompressedTexture2D = (gl, images) => {
```
Comment (belom88) [inline src/components/tile-details-panel/texture-section.tsx:186]: Add TSDoc and types

File: src/components/tile-details-panel/texture-section.tsx [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+  });
+  const [previewSize, setPreviewSize] = useState<SIZE>({
+    width: PREVIEW_AREA_SIZE,
+    height: PREVIEW_AREA_SIZE,
+  });
+
+  const contentImage =
+    tile.content?.material?.pbrMetallicRoughness?.baseColorTexture?.texture
+      ?.source?.image;
+
```
Comment (belom88) [inline src/components/tile-details-panel/texture-section.tsx:342]:
```suggestion
```

File: src/components/tile-details-panel/texture-section.tsx [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+  const originalTexture = tile.userData?.originalTexture;
+  const image = originalTexture || contentImage;
+
+  useEffect(() => {
+    if (image) {
+      if (image.compressed) {
+        drawCompressedTexture(image, AREA_SIZE).then((ret) => {
```
Comment (belom88) [inline src/components/tile-details-panel/texture-section.tsx:349]:
```suggestion
        drawCompressedTexture(image, AREA_SIZE).then((result) => {
```

File: src/components/tile-details-panel/texture-section.tsx [hunk 5]
```diff
@@ -0,0 +1,418 @@
...
+    case COMPRESSED_SRGB_ALPHA_S3TC_DXT1_EXT:
+    case COMPRESSED_SRGB_ALPHA_S3TC_DXT3_EXT:
+    case COMPRESSED_SRGB_ALPHA_S3TC_DXT5_EXT:
+      return supportedFormats.has("dxt-srgb");
+    default:
+      return false;
+  }
+};
+
+type SIZE = {
```
Comment (belom88) [inline src/components/tile-details-panel/texture-section.tsx:157]: Caps is for constants

```suggestion
type Size = {
```

File: src/components/tile-details-panel/texture-section.tsx [hunk 6]
```diff
@@ -0,0 +1,418 @@
...
+  tile: Tile3D;
+};
+
+export const TextureSection = ({ tile }: TextureSectionProps) => {
+  const [texture, setTexture] = useState<string>("");
+  const [previewTexture, setPreviewTexture] = useState<string>("");
+  const [showPreviewTexture, setShowPreviewTexture] = useState<boolean>(false);
+
+  const [size, setSize] = useState<SIZE>({
+    width: AREA_SIZE,
```
Comment (belom88) [inline src/components/tile-details-panel/texture-section.tsx:331]: Confusing naming. "Area" is something equal to `width`*`height`

```suggestion
    width: SIZE,
```

**Review by belom88 — CHANGES_REQUESTED** [review]

File: src/components/modal-dialog/modal-dialog.tsx [hunk 7]
```diff
@@ -88,6 +88,10 @@ const ButtonsContainer = styled.div<ButtonsContainerProps>`
   }
 `;
 
+const AfterButtonsContainer = styled.div`
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/modal-dialog/modal-dialog.tsx]: Done.

File: src/components/tile-details-panel/texture-section.tsx (hunk 2 above)
Comment (mspivak-actionengine) [reply src/components/tile-details-panel/texture-section.tsx]: Done.

File: src/components/tile-details-panel/texture-section.tsx (hunk 3 above)
Comment (mspivak-actionengine) [reply src/components/tile-details-panel/texture-section.tsx]: Done.

File: src/components/tile-details-panel/texture-section.tsx (hunk 4 above)
Comment (mspivak-actionengine) [reply src/components/tile-details-panel/texture-section.tsx]: Done.

File: src/components/tile-details-panel/texture-section.tsx (hunk 5 above)
Comment (mspivak-actionengine) [reply src/components/tile-details-panel/texture-section.tsx]: Done.

File: src/utils/debug/texture-render-utils.ts [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+};
+
+/**
+ * Draws and rescales the compressed image provided
+ * @param data - object containing the compressed image
+ * @param size - size of the image drawn
+ * @returns texture drawn and its size
+ */
+export const drawCompressedTexture = async (
+  data: any,
```
Comment (belom88) [inline src/utils/debug/texture-render-utils.ts:290]: Add types for `data`

**Review by belom88 — APPROVED** [review]

File: src/utils/debug/texture-render-utils.ts (hunk 8 above)
Comment (mspivak-actionengine) [reply src/utils/debug/texture-render-utils.ts]: Done

**Pull request merged**